app/models/channel.rb

```diff
@@ -17,10 +17,16 @@ class Channel < ApplicationRecord
 
   accepts_nested_attributes_for :details
 
+  validates :publisher, presence: true
+
+  validates :details, presence: true
+
   validate :details_not_changed?
 
   validates :verification_status, inclusion: { in: %w(started failed) }, allow_nil: true
 
+  validate :site_channel_details_brave_publisher_id_unique_for_publisher, if: -> { details_type == 'SiteChannelDetails' }
```

Comment:
minor: it feels a little funny to have SiteChannelDetail specific validations on the general Channel